addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/network/Requestor.java

```diff
@@ -60,7 +60,16 @@ public Requestor(int initiator) {
                 HttpMessage httpRequest = new HttpMessage(new URI(url, false));
                 httpRequest.getRequestHeader().setMethod(requestModel.getMethod().name());
                 for (HttpHeaderField hhf : requestModel.getHeaders()) {
-                    httpRequest.getRequestHeader().setHeader(hhf.getName(), hhf.getValue());
+                    if (hhf.getValue().contains("multipart/form-data")) {
+                        String requestBody = requestModel.getBody();
+                        String boundary = requestBody.substring(2, 38);
+                        httpRequest
+                                .getRequestHeader()
+                                .setHeader(
+                                        hhf.getName(), hhf.getValue() + "; boundary=" + boundary);
+                    } else {
+                        httpRequest.getRequestHeader().setHeader(hhf.getName(), hhf.getValue());
+                    }
```

Comment:
This should be reverted, it's not responsibility of the `Requestor` to tweak/correct generated data.

Reply:
@thc202 Should I address this in `RequestModelConverter` then?

Reply:
Yes.